src/qililab/platform/platform.py

```diff
@@ -574,13 +574,34 @@ def execute_qprogram(  # pylint: disable=too-many-locals
         calibration: Calibration | None = None,
         debug: bool = False,
     ) -> QProgramResults:
-        """Execute a QProgram using the platform instruments.
+        """Execute a :class:`.QProgram` using the platform instruments.
+
+        |
+
+        **The execution is done in the following steps:**
+
+        1. Compile the QProgram.
+        2. Run the compiled QProgram.
+        3. Acquire the results.
+
+        |
+
+        **The execution is done in the following way:**
+
+        - If all the buses in the :class:`.QProgram` are associated with Qblox modules, the compilation is done using the :class:`.QbloxCompiler`.
+        - If all the buses in the :class:`.QProgram` are associated with Quantum Machines Clusters, the compilation is done using the :class:`.QuantumMachinesCompiler`.
+        - If the buses in the :class:`.QProgram` are associated with a mixture of Qblox modules and Quantum Machines Clusters, the compilation is not supported.
 
         Args:
-            qprogram (QProgram): The QProgram to execute.
+            qprogram (QProgram): The :class:`.QProgram` to execute.
+            bus_mapping (dict[str, str], optional): A dictionary mapping the buses in the :class:`.QProgram` (keys )to the buses in the platform (values).
+                It is useful for mapping specific experiments to generic :class:`.QProgram` 's. Defaults to None.
+            calibration (Calibration, optional): :class:`.Calibration` file containing information of the experiment in the YAML format. Defaults to None.
+            debug (bool, optional): Whether to print debug information. Defaults to False.
```

Comment:
I would expand the debug a bit
```suggestion
            debug (bool, optional): Whether to create debug information. For QBlox cluster all the program information is printed on screen. For Quantum Machines cluster a .py file is created containing the QUA and config compilation. Defaults to False.
```